### PR TITLE
MB-16796 Correct issue where name and date data don't populate on the legal agreement

### PR DESCRIPTION
**Pull request by mandaem**
## Summary

This fix should update the Legal PDF to print the customer’s name and the date of the customer’s signing as presented when the customer sets up their shipment and signing. Check out 'How to test' below for details.

If we should limit to certain statuses, it definitely should be more than just submitted. Is there any problem with just not limiting to any status in that conditional?

pdf still has no sign/date (at least before step 5 is done). Is that expected functionality for PPM's pdf?

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the customer app in ephemeral (and local if you'd like too, I didn't notice any difference between them but please let me know if you do).
2. Create a new user with that button and make sure it allows you to see sign/date in the pdf after you've submitted everything
3.  use your login.gov to make move(s) and make sure it allows you to see sign/date in the pdf after you've submitted everything
4. you can also check a lot of the test data users' pdfs of various types of moves too to see if they're signed/dated.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/about/supported-browsers) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/about/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Comment by mandaem** [comment]
Checked it's both saving and retrieving data no matter what type of user (test based, our logins) or environment. There were problems with sign/date displaying sometimes in local and sometimes not, so the issues are not tied only to stg and ephemeral environments. That's how I figured out the issue was ultimately that when it was showing signature and date when it was in submitted status, and everything else it wouldn't. The problem is, when it's in other statuses (like 'needs service counseling'), it should be allowing the user to print the agreement with their signature and date displayed. So we can't keep to just 'submitted'. I remedied this by removing  `move.status === MOVE_STATUSES.SUBMITTED `. conditional in  [componentDidMount()](url) Now those fields from populating for things other than 'submitted'.  Duncan noted that experimental is going to be quite busy but to make sure a fix works in ephemeral, I'll double check this in ephemeral too, but this seems to be the issue.

**Comment by mandaem** [comment]
Confirmed changes in signature and date being applied now in ephemeral too. Tested many scenarios, where they were in the process, test data, new user, logged in, etc. Seems to be functioning correctly now for HHG, but will test for other types too.

**Comment by mandaem** [comment]
I confirmed its showing the signature and date being applied for NTS and combo moves too.

**Comment by mandaem** [comment]
After more testing, if step 5 for solo PPMs is completed (including asking for signature), then it appears the pdf will show sign and date now (and if anyone sees otherwise on what I'm reporting on the PPMs --or any of the comments I've made-- please let me know). Whereas before my change, they didn't show even when step 5 was completed.

So pdf looks to have the sign and date when step 5/uploaded PPM docs is completed, but not before that step's done. Is that expected functionality for PPM's pdf?

Also for combo PPM/HHG ones, it does show sign and date before step 5 is complete.

**Comment by mandaem** [comment]
After more testing, if step 5 for solo PPMs is completed (including asking for signature), then it seems to appear the pdf will show sign and date now (and if anyone sees otherwise on what I'm reporting on the PPMs --or any of the comments I've made-- please let me know). Whereas before my change, they didn't show even when step 5 was completed.

So pdf looks to have the sign and date when step 5/uploaded PPM docs is completed, but not before that step's done. Is that expected functionality for PPM's pdf?

Also for a combo PPM/HHG one, it did show sign and date before step 5 is complete.

**Review by PDickeyTruss — APPROVED** [review]
Looks reasonable to me, awesome work tracking this down! I can't think of a good reason not to show the signature once the doc has been signed and the existence of the signature to me means the doc was signed. 🤷  Approving with the disclaimer that I have no knowledge of why the status filtering was added originally, so we may want some additional eyes/approvals on this before merging.

**Review by brandonlenz — APPROVED** [review]
Confirmed this works in ephemeral (and local) following the testing steps!

I can't speak to some of the questions around PPMs and potential intentionality of _not_ showing the signature. It seems like something there would be less of a business reason to do, and more of a race condition. If someone was able to move through the UX such that they were able to sign the document, we should show that signature, regardless of the status of their move, IMO, rather than silently not showing it.

**Comment by duncan-truss** [comment]
I don't know what the best value is for our time here in terms of implementing testing:

* Testing that the getSignedCertification is called on render (seems achievable, [display will still be set to none on the printable media style](https://github.com/transcom/mymove/blob/8263d616ca3775b25acab11d60efdada02286cba/src/components/Customer/Home/PrintableLegalese/PrintableLegalese.module.scss#L2) unless we take other action)
* Testing that clicking on the print link calls our window.print method (a little harder, setting up a mock on the global window object.  Doesn't give us much more confidence)
* Mocking the response of getSignedCertification and applying the print media style to the component in testing-library (not simple but feels like the right test) https://www.js-howto.com/test-responsive-design-using-jest-and-react-testing-library/
* Similar to the last one but with Playwright (will require more work on seed data creation and unclear what the end result would be)

@PDickeyTruss curious on your thoughts here

**Comment by PDickeyTruss** [comment]
Re testing: 
My instinct is just to mock getSignedCertification (`jest.fn()`), then just assert it was called after rendering the component with RTL. The status check that was removed would have prevented this from getting called, so that does feel like we are testing the change that was made.

Also, I wonder if we have any existing playwright tests that view the legal agreement as a part of their flow? I suspect not, but if so maybe we could get away with just adding a few assertions around the signature/date being shown (might be a little tricky coming up with good selectors). Other than that

**Comment by mandaem** [comment]
Thanks all. Looks like there aren't scenarios where the legal agreement is part of the flow from what I see https://github.com/transcom/mymove/tree/main/playwright/tests

**Comment by duncan-truss** [comment]
> Thanks all. Looks like there aren't scenarios where the legal agreement is part of the flow from what I see https://github.com/transcom/mymove/tree/main/playwright/tests

@mandaem That seems accurate otherwise we might have caught this depending on the test data used.  How does focusing on adding a client test to verify that the getSignedCertification API gets called sound?

**Comment by mandaem** [comment]
> @mandaem That seems accurate otherwise we might have caught this depending on the test data used. How does focusing on adding a client test to verify that the getSignedCertification API gets called sound?

I can do that if you'd like, Duncan. I was looking at mockGetMoveDocumentsForMove in src/scenes/Moves/Ppm/PaymentReview/DocumentsUploaded.test.js
@PDickeyTruss @duncan-truss  is this a good way to approach for getsignedcert or is there a better example?

**Comment by duncan-truss** [comment]
> > @mandaem That seems accurate otherwise we might have caught this depending on the test data used. How does focusing on adding a client test to verify that the getSignedCertification API gets called sound?
> 
> I can do that if you'd like, Duncan. I was looking at mockGetMoveDocumentsForMove in src/scenes/Moves/Ppm/PaymentReview/DocumentsUploaded.test.js @PDickeyTruss @duncan-truss is this a good way to approach for getsignedcert or is there a better example?

Yep it can be similar to that example or in this component's index.test.jsx you can see there are example assertions for `mockNavigate` being called with the expected arguments when a link is clicked https://jestjs.io/docs/mock-functions

**Comment by PDickeyTruss** [comment]
I think the example in DocumentsUploaded.test.js is similar to what you want to do. It looks like [getSignedCertification](https://github.com/transcom/mymove/blob/main/src/pages/MyMove/Home/index.jsx#L575) is just a function passed in as a prop to the (MyMove) Home component. We do actually use redux for this prop but hopefully that wont matter for this test as you will mocking the whole function. If you checkout the Home component tests you can see they actually already mock this function, just have no assertions related to it: https://github.com/transcom/mymove/blob/main/src/pages/MyMove/Home/index.test.jsx#L51 

I think if I were writing this test I would do something like this in the MyMove Home component tests:
```Javascript
const mockGetSignedCertification = jest.fn()

// render Home component with mocked function
render(
  <MockProviders>
    <Home {...defaultProps} {...props} getSignedCertification={mockGetSignedCertification} />
  </MockProviders>
);

// Assert things
expect(mockGetSignedCertification).toHaveBeenCalledTimes(1)
expect(mockGetSignedCertification).toHaveBeenWith(moveId) // The move id is currently generated each time so you may need to make it constant or remove this assertion: https://github.com/transcom/mymove/blob/main/src/pages/MyMove/Home/index.test.jsx#L61
```

File: src/pages/MyMove/Home/index.test.jsx [hunk 1]
```diff
@@ -331,6 +335,25 @@ describe('Home component', () => {
     });
   });
 
+  describe('if user has submitted PPM with signed agreement', () => {
+    const props = {
+      mtoShipments: [createSubmittedPPMShipment()],
```
Comment (PDickeyTruss) [inline src/pages/MyMove/Home/index.test.jsx:340]: Good call doing this with a move in submitted state to fit the conditional logic that was changed 🎉

File: src/pages/MyMove/Home/index.test.jsx [hunk 2]
```diff
@@ -331,6 +335,25 @@ describe('Home component', () => {
...
+      orders,
+      uploadedOrderDocuments,
+    };
+
+    const mockGetSignedCertification = jest.fn();
```
[the remark this answers is not in the document]
Comment (mandaem) [reply src/pages/MyMove/Home/index.test.jsx]: done :)

**Review by PDickeyTruss — APPROVED** [review]
LGTM! Code change fixes the issue and there is a unit test verifying the change looks good 🎉

**Pull request merged**